### ISSUE TITLE
Add a filter to remove certain types of code lines from the count

### ISSUE THREAD

**Issue by EnricoPicci** [AlDanial/cloc#868]
Would it be possible to add a mechanism to filter from the "lines of code" count code lines that match a certain pattern?
For instance, in Java I would like to remove the changes that affect "import" lines.

**Comment by AlDanial** [comment]
Docs for the new switch `--ignore-regex`:
<pre>
   --ignore-regex            Ignore lines in source files that match the given                                                    
                             Perl regular expression for the given language(s).  
                             This option can be specified multiple times.                                                         
                             Language names are comma separated and are followed                                                  
                             by the pipe character and the regular expression.                                                    
                             Use * to match all languages.                                                                        
                             Examples:                                                                                            
                               --ignore-regex="C,Java,C++|^\s*[{};]\s*$"                                                          
                               --ignore-regex="*|DEBUG|TEST\s+ONLY"                                                               
                             These filters are applied after comments are                                                         
                             removed.  Use --strip-comments=EXT to create     
                             new files that show these filters applied.   
</pre>

**Comment by AlDanial** [comment]
Sure, `--ignore-regex` will be made to work with `--diff`,  that was my original intention, but wanted to implement this new feature in steps since it was relatively involved.
I have other time commitments so it will be a couple of weeks before I get to this though.

**Comment by EnricoPicci** [comment]
Thanks a lot, your work is great, I am looking forward trying --ignore-regex with diff. Have a great day.

**Comment by AlDanial** [comment]
That's right, I was waiting for you to confirm the combination of `--ignore-regex` and `--diff`.
